### PR TITLE
Add new `AdwButton` styles

### LINKED ISSUE #48: Update `AdwSidebar` item default colors
The colors used inside the `AdwSidebar` do not match the ones from GTK's implementation.

### PR DESCRIPTION
**Pull request by jesusrp98**
Closes #48 and #49.

This PR updates the `AdwButton` widget with new default styles, such as flat, pill and circle.

image

Here's a list of all changes implemented:
- Adapt default `AdwButton` constructor to represent the 'regular' adwaita button.
- Introduce `AdwButton.circular` style.
- Introduce `AdwButton.pill` style.
- Introduce `AdwButton.flat` style.
- Deprecate `AdwTextButton` in favor of `AdwButton`.
- Introduce `child` parameter on `AdwButton`, that is passed to the `builder` parameter (similar to other widgets in Flutter).
- Update animation duration and border radius parameters to coincide with `libadwaita`.
- Update all button state color to coincide with `libadwaita`.
- Add `textStyle` parameter to `AdwButton`.
- Simplify `AdwHeaderButton` and `AdwSidebar` widgets.

### CODE REVIEW

**Review by prateekmedia — APPROVED** [review]
Looks good to me!

**Comment by MalcolmMielle** [comment]
I can have a look later tonight if you want me to @jesusrp98? Up to you.

File: lib/src/widgets/adw/button.dart [hunk 1]
```diff
@@ -163,10 +279,93 @@ class _AdwButtonState extends State<AdwButton> {
...
+
+extension AdwButtonBackgroundColor on Color {
+  Color? resolveDefaultAdwButtonColor(
+    BuildContext context,
+    AdwButtonStatus status,
+  ) {
+    if (Theme.of(context).brightness == Brightness.light) {
+      switch (status) {
+        case AdwButtonStatus.enabled:
+          return withOpacity(0.08);
```
Comment (MalcolmMielle) [inline lib/src/widgets/adw/button.dart:309]: Today I learned, I didn't that that was doable with extension :D. That's cool

**Review by MalcolmMielle — APPROVED** [review]
Super cool PR! Thanks

**Comment by MalcolmMielle** [comment]
For the error given by the linter, I've open #51 because the super-linter ended up being quite annoying :S. I'm sorry, I know I introduced it, but I thought it would be nicer than it is :(.

**Pull request merged**